### PR TITLE
Windows build

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: gen_api.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        continue;
+    }
+
+    // Strip macros/includes
+    if (preg_match('/^#/', $line)) {
+        continue;
+    }
+
+    // Strip visibility attributes
+    $line = preg_replace('/AWS_CRT_API /', '', $line);
```
Comment (SamRemis) [inline gen_api.php]: str_replace is faster if you don't need regex

File: gen_api.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+$input_lines = file($source);
+foreach ($input_lines as $line) {
+    // Strip AWS_EXTERN_C_BEGIN/END
+    if (preg_match('/AWS_EXTERN_C/', $line)) {
+        continue;
+    }
+
+    // Strip macros/includes
+    if (preg_match('/^#/', $line)) {
```
Comment (SamRemis) [inline gen_api.php:16]: this isn't actually stripping anything, right? it's just saying 
//if it matches the regex
// continue

Comment (justinboswell) [reply gen_api.php:16]: Correct! It's to strip the lines from the output.

File: gen_api.php (hunk 1 above)
Comment (justinboswell) [reply gen_api.php]: Fixed!

File: gen_api.php (hunk 2 above)
Comment (SamRemis) [reply gen_api.php:16]: Oh right!  I spaced

File: Makefile.frag [hunk 3]
```diff
@@ -1,8 +1,5 @@
 
-BUILD_DIR=$(shell pwd)/build
-DEPS_DIR=$(BUILD_DIR)/deps
-INT_DIR=$(BUILD_DIR)/install
-INSTALL_DIR=$(shell pwd)
+INT_DIR=build/install
```
Comment (justinboswell) [inline Makefile.frag:2]: BUILD_DIR is used by the global Makefile, and this was confusing make

File: Makefile.frag [hunk 4]
```diff
@@ -11,7 +8,7 @@ ifeq (, $(shell which cmake3))
 	CMAKE = cmake
 endif
 
-CMAKE_CONFIGURE = $(CMAKE) -DCMAKE_INSTALL_PREFIX=$(INT_DIR) -DCMAKE_PREFIX_PATH=$(INT_DIR) -DBUILD_TESTING=OFF -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE)
+CMAKE_CONFIGURE = $(CMAKE) -DCMAKE_INSTALL_PREFIX=$(INT_DIR) -DBUILD_TESTING=OFF -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE)
```
Comment (justinboswell) [inline Makefile.frag:11]: PREFIX_PATH is not necessary because we're not building against extant dependencies

File: Makefile.frag [hunk 5]
```diff
@@ -43,7 +40,7 @@ endif
 
 # transform/install api.h from FFI lib
 src/api.h: crt/aws-crt-ffi/src/api.h
-	cat crt/aws-crt-ffi/src/api.h | grep -v AWS_EXTERN_C | sed -e 's/AWS_CRT_API //' | grep -ve '^#' > src/api.h
+	php gen_api.php crt/aws-crt-ffi/src/api.h > src/api.h
```
Comment (justinboswell) [inline Makefile.frag:43]: Moved this to a script so the same script can be used on windows

**Review by SamRemis — APPROVED** [review]

**Pull request merged**